syft/pkg/cataloger/python/package.go

```diff
@@ -123,3 +136,85 @@ func vcsURLQualifierForPackage(p *pkg.PythonDirectURLOriginInfo) packageurl.Qual
 		{Key: pkg.PURLQualifierVCSURL, Value: fmt.Sprintf("%s+%s@%s", p.VCS, p.URL, p.CommitID)},
 	}
 }
+
+func enrichLicenseIfConfigured(ctx context.Context, cfg CatalogerConfig, name string, version string) pkg.LicenseSet {
+	var licenseSet pkg.LicenseSet
+
+	if cfg.SearchRemoteLicenses {
+		license, err := getLicenseFromPypiRegistry(cfg.PypiBaseURL, name, version)
+		if err == nil && license != "" {
+			licenses := pkg.NewLicensesFromValuesWithContext(ctx, license)
+			licenseSet = pkg.NewLicenseSet(licenses...)
+		}
+		if err != nil {
+			log.Debugf("unable to extract licenses from pypi registry for package %s:%s: %+v", name, version, err)
+		}
+	}
+	return licenseSet
+}
+
+func formatPypiRegistryURL(baseURL, packageName, version string) (requestURL string, err error) {
+	urlPath := []string{packageName, version, "json"}
+	requestURL, err = url.JoinPath(baseURL, urlPath...)
+	if err != nil {
+		return requestURL, fmt.Errorf("unable to format pypi request for pkg:version %s%s; %w", packageName, version, err)
+	}
+	return requestURL, nil
+}
+
+func getLicenseFromPypiRegistry(baseURL, packageName, version string) (string, error) {
+	// "https://pypi.org/pypi/%s/%s/json", packageName, version
+	requestURL, err := formatPypiRegistryURL(baseURL, packageName, version)
```

Comment:
One thing I missed with the prior PR, since I think it was using an existing function, is that we have a simple mechanism to cache results we should use for remote fetches like this, for example [golang license lookups use this caching](https://github.com/anchore/syft/blob/07029ead8a2c6ba727757e494b516ea740b210e2/syft/pkg/cataloger/golang/licenses.go#L174). We should probably use this functionality for any new remote fetches. (And of course not as part of this PR, but retrofit any existing fetching.) Aside from this detail, everything else looks great to me here -- especially you've made sure not to break the public API, which is 👍

Reply:
@kzantow I think I addressed this with the most recent commit. Happy to add the license caching to the JS stuff after this PR and https://github.com/anchore/syft/pull/4284 gets merged. Appreciate the reviews!